From a21bb26dcfcc00425f031421b87576f9c81e4824 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 24 Jan 2024 20:44:29 +0100
Subject: [PATCH 21/21] libcamera: debayer_cpu: Make the minimum size 1280x720

pipewire + firefox default to what looks like 640x480 if we export
the entire supported cropping range. Hardcode 720p as minsize for now.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 src/libcamera/software_isp/debayer_cpu.cpp | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/src/libcamera/software_isp/debayer_cpu.cpp b/src/libcamera/software_isp/debayer_cpu.cpp
index d6599805..5a06b191 100644
--- a/src/libcamera/software_isp/debayer_cpu.cpp
+++ b/src/libcamera/software_isp/debayer_cpu.cpp
@@ -790,10 +790,17 @@ SizeRange DebayerCpu::sizes(PixelFormat inputFormat, const Size &inputSize)
 		return {};
 	}
 
-	return SizeRange(Size(pattern_size.width, pattern_size.height),
-			 Size((inputSize.width - 2 * pattern_size.width) & ~(pattern_size.width - 1),
-			      (inputSize.height - 2 * border_height) & ~(pattern_size.height - 1)),
-			 pattern_size.width, pattern_size.height);
+	/*
+	 * pipewire + firefox default to what looks like 640x480
+	 * if we export the entire supported cropping range.
+	 * Hardcode 720p as minsize for now. Minsize should be
+	 * Size(pattern_size.width, pattern_size.height)
+	 */
+	unsigned int w = (inputSize.width - 2 * pattern_size.width) & ~(pattern_size.width - 1);
+	unsigned int h = (inputSize.height - 2 * pattern_size.height) & ~(pattern_size.height - 1);
+	return SizeRange(Size(std::min(w, 1280u), std::min(h, 720u)),
+	                 Size(w, h),
+	                 pattern_size.width, pattern_size.height);
 }
 
 } /* namespace libcamera */
-- 
2.43.2

